Model/Resolver/PlaceRazorpayOrder.php

```diff
@@ -0,0 +1,96 @@
+use Razorpay\Magento\Model\PaymentMethod;
+
+class PlaceRazorpayOrder implements ResolverInterface
+{
+    protected $scopeConfig;
+
+    protected $cartManagement;
+
+    protected $_objectManager;
+
+    public function __construct(
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
+        GetCartForUser $getCartForUser,
+        \Magento\Quote\Api\CartManagementInterface $cartManagement,
+        PaymentMethod $paymentMethod
+    ) {
+        $this->scopeConfig    = $scopeConfig;
+        $this->getCartForUser = $getCartForUser;
+        $this->cartManagement = $cartManagement;
+        $this->rzp = $paymentMethod->rzp;
+        $this->_objectManager   = \Magento\Framework\App\ObjectManager::getInstance();
+    }
+
+    /**
+     * @param GetCartForUser $getCartForUser
+     * @inheritdoc
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
+    {
+        if (empty($args['order_id'])) {
+            throw new GraphQlInputException(__('Required parameter "order_id" is missing'));
+        }
+        try {
+            $storeScope = \Magento\Store\Model\ScopeInterface::SCOPE_STORE;
+            $storeId = (int) $context->getExtensionAttributes()->getStore()->getId();
+            $receipt_id        = $args['order_id'];
+            $collection = $this->_objectManager->get('Magento\Sales\Model\Order')
+            ->getCollection()
+            ->addFieldToSelect('*')
+            ->addFilter('increment_id', $receipt_id)
+            ->getFirstItem();
+            $salesOrder = $collection->getData();
+            $amount          = (int) (number_format($salesOrder['grand_total'] * 100, 0, ".", ""));
+            $payment_action  = $this->scopeConfig->getValue('payment/razorpay/payment_action', $storeScope);
+            $payment_capture = 1;
+            if ($payment_action === 'authorize') {
+                $payment_capture = 0;
+            }
+            $order = $this->rzp->order->create([
+                'amount'          => $amount,
+                'receipt'         => $receipt_id,
+                'currency'        => $salesOrder['order_currency_code'],
+                'payment_capture' => $payment_capture,
+                'app_offer'       => (($salesOrder['grand_total'] - $salesOrder['base_discount_amount']) > 0) ? 1 : 0,
+            ]);
+            if (null !== $order && !empty($order->id)) {
+
+                $responseContent = [
+                    'success'        => true,
+                    'rzp_order_id'   => $order->id,
+                    'order_quote_id' => $receipt_id,
+                    'amount'         => number_format((float) $salesOrder['grand_total'], 2, ".", ""),
+                    'currency'       => $salesOrder['order_currency_code'],
+                    'message'        => 'Razorpay Order created successfully'
+                ];
+                return $responseContent;
```

Comment:
@enishant , we needs to store `rzp_order_id` on server.

Reply:
@ramth05 stored rzp_order_id to database at table sales_order

Reply:
@enishant , looks like we are missing logger here as well.

Reply:
@ramth05 Added logger to PlaceRazorpayOrder & setRazorpayPaymentDetailsforOder